### ISSUE TITLE
eldoc for Python includes 'self' argument

### ISSUE THREAD

**Issue by piojanu** [Valloric/ycmd#1066]
Hi!

I've just started using ycmd and I see that ycmd eldoc for python includes 'self' argument in its print to echo area while hovering over class method. It's unwanted as indication in bold of which argument we are currently typing is shifter to left by one (as you all know, you don't pass self argument explicitly).

You can see it in here:
<img width="634" alt="screen shot 2018-07-16 at 18 21 59" src="https://user-images.githubusercontent.com/23562932/42770569-370bfe3a-8925-11e8-87f2-f0093de18ff3.png">

It feels like it's a bug. Can I change it somehow to not to include it? anaconda-mode-eldoc was showing it just fine, but I moved from it as I need something with multi-language support.

Thanks for your time!

**Comment by bstaletic** [comment]
I'm not sure what "eldoc" is, but it's not a part of ycmd. Ycmd doesn't echo a function's signature to the command line. What YouCompleteMe does do is open a preview buffer in order to display such information.

**Comment by puremourning** [comment]
I suspect you need to raise this with emacs-ycmd

**Comment by piojanu** [comment]
@puremourning That's what I did (see reference two positions above), they told me that they print function signature they get from server, so it might be a fix on server-side. If you deny that, I'll come back to them.

**Comment by bstaletic** [comment]
Alright, this is actually what I've ran into when I tried to hack snippet epansion of function signatures in YCM. 

What ycmd is doing is:
- Issuing a `/completions` request to Jedi.
- Taking the `'completions'` part of the response.
- Iterating over the `completions` list and generating the completions like this:
  - Jedi `completion['name']` becomes ycmd `insertion_text`
  - Jedi `completion['description']` becomes ycmd `extra_menu_info`.
  - Jedi `completion['docstring']` becomes ycmd `detailed_info` (the one I parsed for thte snippets).
  - Possibly some `extra_data` is simply passed through.

So ycmd isn't doing any processing of what jedi sends, rather ycmd just repacks the data into its own format.
